client/azure.go

```diff
@@ -22,12 +22,21 @@ type AzureConfig struct {
 	ClientID string
 	// ClientSecret is the secret of the Active Directory Service Principal.
 	ClientSecret string
+	// The cloud environment identifier. Takes values from https://github.com/Azure/go-autorest/blob/ec5f4903f77ed9927ac95b19ab8e44ada64c1356/autorest/azure/environments.go#L13
+	Cloud string
 	// SubscriptionID is the ID of the Azure subscription.
 	SubscriptionID string
 	// TenantID is the ID of the Active Directory tenant.
 	TenantID string
 }
 
+// azureClientConfig contains all essential information to create an Azure client.
+type azureClientConfig struct {
+	subscriptionID          string
+	resourceManagerEndpoint string
+	servicePrincipalToken   *adal.ServicePrincipalToken
+}
```

Comment:
Made this similar to https://github.com/kubernetes/kubernetes/blob/v1.10.2/pkg/cloudprovider/providers/azure/azure_client.go#L146-L155

Idea is to get oAuth token once (put to this object) and use it for clients.